gui/dig.lua

```diff
@@ -1317,23 +1676,104 @@ function Dig:commit()
         self.placing_extra = { active = false, index = nil }
         self.extra_points = {}
         self.prev_center = nil
+        self.start_center = nil
     end
+
     self:updateLayout()
 end
 
+function Dig:get_mirrored_points(points)
+    local mirror_horiz_value = self.subviews.mirror_horiz_label:getOptionValue()
+    local mirror_diag_value = self.subviews.mirror_diag_label:getOptionValue()
+    local mirror_vert_value = self.subviews.mirror_vert_label:getOptionValue()
+
+    local mirrored_points = {}
+    for i = #points, 1, -1 do
+        local point = points[i]
+        -- 1 maps to "Off"
+        if mirror_horiz_value ~= 1 then
+            local mirrored_y = self.mirror_point.y + ((self.mirror_point.y - point.y))
+
+            -- if Mirror (even), then increase mirror amount by 1
+            if mirror_horiz_value == 3 then
+                if mirrored_y > self.mirror_point.y then
+                    mirrored_y = mirrored_y + 1
+                else
+                    mirrored_y = mirrored_y - 1
+                end
+            end
+
+            table.insert(mirrored_points, { z = point.z, x = point.x, y = mirrored_y })
+        end
+    end
+
+    for i, point in ipairs(points) do
+        if mirror_diag_value ~= 1 then
+            local mirrored_y = self.mirror_point.y + ((self.mirror_point.y - point.y))
+            local mirrored_x = self.mirror_point.x + ((self.mirror_point.x - point.x))
+
+            -- if Mirror (even), then increase mirror amount by 1
+            if mirror_diag_value == 3 then
+                if mirrored_y > self.mirror_point.y then
+                    mirrored_y = mirrored_y + 1
+                    mirrored_x = mirrored_x + 1
+                else
+                    mirrored_y = mirrored_y - 1
+                    mirrored_x = mirrored_x - 1
+                end
+            end
+
+            table.insert(mirrored_points, { z = point.z, x = mirrored_x, y = mirrored_y })
+        end
+    end
+
+    for i = #points, 1, -1 do
+        local point = points[i]
+        if mirror_vert_value ~= 1 then
+            local mirrored_x = self.mirror_point.x + ((self.mirror_point.x - point.x))
+
+            -- if Mirror (even), then increase mirror amount by 1
+            if mirror_vert_value == 3 then
+                if mirrored_x > self.mirror_point.x then
+                    mirrored_x = mirrored_x + 1
+                else
+                    mirrored_x = mirrored_x - 1
+                end
+            end
+
+            table.insert(mirrored_points, { z = point.z, x = mirrored_x, y = point.y })
+        end
+    end
+
+    for i, point in ipairs(mirrored_points) do
+        table.insert(points, mirrored_points[i])
+    end
+
+    return points
+end
+
 --
--- Dig
+-- DigScreen
 --
 
 DigScreen = defclass(DigScreen, gui.ZScreen)
 DigScreen.ATTRS {
     focus_path = "dig",
     pass_pause = true,
     pass_movement_keys = true,
+    dig_window = DEFAULT_NIL,
+    debug_window = DEFAULT_NIL
```

Comment:
if they're not being set by a parent, these don't need to be class attributes. they can be initialized as self.dig_window and self.debug_window in init()